add additional arguments to `publish-sdk`

jpculp opened the pull request:
**Description of changes:**

Adds the ability to skip publishing a multi-arch manifest (`--skip-manifest`) or publish a multi-arch manifest with remote images that were pushed beforehand (`--only-manifest`).

Also adds a counter for how many ECR operations _(container/manifest pushes)_ occurred and compare it to the expected minimum number of operations. If the count is lower than expected _(ex. nothing got pushed)_, the script will return a 1.


**Testing done:**
- Built SDK
- Built SDK with SKIP_MANIFEST="true"
- Built SDK with ONLY_MANIFEST="true"

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

bcressey commented on publish-sdk:
```diff
@@ -10,13 +10,21 @@ for opt in "$@"; do
       --registry=*) REGISTRY="${optarg}" ;;
       --sdk-name=*) SDK_NAME="${optarg}" ;;
       --short-sha=*) SHORT_SHA="${optarg,,}" ;;
+      --skip-manifest=*) SKIP_MANIFEST="${optarg,,}" ;;
+      --only-manifest=*) ONLY_MANIFEST="${optarg,,}" ;;
```
Can we error out if `SKIP_MANIFEST` and `ONLY_MANIFEST` are both set, or exit immediately with "nothing to do" in that case?

bcressey commented on publish-sdk:
```diff
@@ -51,64 +59,70 @@ update_image() {
-
-  if ! docker tag "${local_host_arch_image}" "${remote_host_arch_image}" ; then
-    echo "failed to tag ${remote_host_arch_image}" >&2 && exit 1
+  if [[ "${ONLY_MANIFEST:-"false"}" != "true" ]]; then
```
Might be good to add the `u` option to error on unset variables, then default both of the new ones to "false" at the top of the script.

jpculp replied:
I had considered that, but I wanted to allow for **ONLY_MANIFEST** and **SKIP_MANIFEST** to be manipulated with environment variables. It's useful for automated build systems like AWS CodeBuild. I could probably still add the `u` option and just set `ONLY_MANIFEST="${ONLY_MANIFEST:-"false"}"` at the top for the best of both worlds.

jpculp replied:
It looks like we already set `u` within the update image function so we can have the nicer required arguments error. I'll still go ahead and set those two variables at the top though.

bcressey commented on publish-sdk:
```diff
@@ -51,64 +59,70 @@ update_image() {
-  # push the manifest and remove the local cache
-  if ! docker manifest push --purge "${remote_image}" ; then
-    echo "failed to push ${remote_image}" >&2 && exit 1
+  if [[ "${SKIP_MANIFEST:-"false"}" != "true" ]]; then
```
You can invert the conditional and just return early if `SKIP_MANIFEST` is "true". That will avoid a lot of the indentation-related changes.

jpculp committed:
add additional arguments to `publish-sdk`

Adds the ability to skip publishing a multi-arch manifest
(`--skip-manifest`) or publish a multi-arch manifest with remote images
that were pushed beforehand (`--only-manifest`).

Signed-off-by: Patrick J.P. Culp <jpculp@amazon.com>

jpculp committed:
check ECR operations in `publish-sdk`

Adds a counter for how many ECR operations (container/manifest pushes)
occurred and compare it to the expected minimum number of operations. If
the count is lower than expected (ex. nothing got pushed), the script
will return a 1.

Signed-off-by: Patrick J.P. Culp <jpculp@amazon.com>

jpculp commented:
- Default to "false" when **SKIP_MANIFEST** or **ONLY_MANIFEST** are unset.
- Exit early if both **SKIP_MANIFEST** and **ONLY_MANIFEST** are both "true".
- Return early if  **SKIP_MANIFEST** is "true".
